src/app/features/file/file.module.ts

```diff
@@ -20,6 +21,7 @@ import { FileDatatype } from "./file.datatype";
         : injector.get(MockFileService);
     }),
     { provide: DefaultDatatype, useClass: FileDatatype, multi: true },
+    { provide: DefaultDatatype, useClass: PhotoDatatype, multi: true },
```

Comment:
I think you still need to register the percentage-datatype also. Let's do that where the NumberDatatype is registered (not here but probably in CoreModule).